e2etests/node_scan_rhcos_test.go

```diff
@@ -0,0 +1,166 @@
+	}{
+		{
+			request: &v1.GetNodeVulnerabilitiesRequest{
+				OsImage:          "Red Hat Enterprise Linux CoreOS 45.82.202008101249-0 (Ootpa)",
+				KernelVersion:    "0.0.1",
+				KubeletVersion:   "0.0.1",
+				KubeproxyVersion: "0.0.1",
+				Runtime: &v1.GetNodeVulnerabilitiesRequest_ContainerRuntime{
+					Name:    "docker",
+					Version: "0.0.1",
+				},
+				NodeInventory: &v1.Components{
+					Namespace: "Namespace-Name",
+					RhelComponents: []*v1.RHELComponent{
+						{
+							Id:        int64(1),
+							Name:      "libksba",
+							Namespace: "rhel:8",
+							Version:   "1.3.5-7.el8",
+							Arch:      "x86_64",
+							Module:    "", // must be empty, otherwise scanner does not return any vulns
+							// From: https://www.redhat.com/security/data/metrics/repository-to-cpe.json
+							// "rhel-8-for-x86_64-appstream-rpms": {"cpes": ["cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream"]},
+							// "rhel-8-for-x86_64-baseos-rpms": {"cpes": ["cpe:/o:redhat:enterprise_linux:8::baseos", "cpe:/o:redhat:rhel:8.3::baseos"]}
+							Cpes: []string{
+								"cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream",
+								"cpe:/a:redhat:enterprise_linux:8::baseos", "cpe:/a:redhat:rhel:8.3::baseos",
+							},
+							AddedBy: "",
+						},
+						{
+							Id:        int64(2),
+							Name:      "tar",
+							Namespace: "rhel:8",
+							Version:   "1.27.1.el8",
+							Arch:      "x86_64",
+							Module:    "",
+							Cpes: []string{
+								"cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream",
+								"cpe:/a:redhat:enterprise_linux:8::baseos", "cpe:/a:redhat:rhel:8.3::baseos",
+							},
+							AddedBy: "",
+						},
+					},
+					LanguageComponents: nil,
+				},
+			},
+			responseContains: &v1.GetNodeVulnerabilitiesResponse{
+				InventoryFeatures: []*v1.Feature{
+					{
+						Name:            "libksba",
+						Version:         "1.3.5-7.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{vulnLibksba},
+					},
+					{
+						Name:            "tar",
+						Version:         "1.27.1.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{vulnTar},
+					},
+				},
+			},
+		},
+	}
+
+	for i, c := range cases {
+		t.Run(fmt.Sprintf("case-%d", i), func(t *testing.T) {
+			c := c
+			resp, err := client.GetNodeVulnerabilities(context.Background(), c.request)
+			require.NoError(t, err)
+			for _, expectedFeat := range c.responseContains.InventoryFeatures {
+				var found bool
+				for _, gotFeat := range resp.InventoryFeatures {
+					if expectedFeat.GetName() == gotFeat.GetName() && expectedFeat.GetVersion() == gotFeat.GetVersion() {
+						found = true
+						assert.NotNil(t, gotFeat)
+						assert.NotNil(t, gotFeat.Vulnerabilities, "Expected to find vulnerabilities for %s:%s", expectedFeat.GetName(), expectedFeat.GetVersion())
+						t.Logf("gotFeat.InventoryFeatures: %+v\n", resp.InventoryFeatures)
```

Comment:
Is this dev-time logging? Looks like it will produce excessive output by dumping the same info over and over.